app/controllers/application_controller.rb

```diff
@@ -0,0 +1,23 @@
+class ApplicationController < ActionController::Base
+  protect_from_forgery with: :exception
+
+  helper Pagy::Frontend
+  helper_method :current_user_order
+
+  def current_user_order
+    order = unfinished_order || Order.find_or_create_by(id: session[:order_id])
```

Comment:
move logic for order into separate service
`Orders::FetchCurrent.new(current_user, session[:order_id])`

Reply:
and add tests for it